processor/transformprocessor/internal/logs/logs.go

```diff
@@ -0,0 +1,466 @@
+			if s, ok := val.(string); ok {
+				ctx.GetItem().(plog.LogRecord).SetSeverityText(s)
+			}
+		},
+	}
+}
+
+func accessBody() pathGetSetter {
+	return pathGetSetter{
+		getter: func(ctx common.TransformContext) interface{} {
+			return getValue(ctx.GetItem().(plog.LogRecord).Body())
+		},
+		setter: func(ctx common.TransformContext, val interface{}) {
+			setValue(ctx.GetItem().(plog.LogRecord).Body(), val)
+		},
+	}
+}
+
+func accessAttributes() pathGetSetter {
+	return pathGetSetter{
+		getter: func(ctx common.TransformContext) interface{} {
+			return ctx.GetItem().(plog.LogRecord).Attributes()
+		},
+		setter: func(ctx common.TransformContext, val interface{}) {
+			if attrs, ok := val.(pcommon.Map); ok {
+				ctx.GetItem().(plog.LogRecord).Attributes().Clear()
+				attrs.CopyTo(ctx.GetItem().(plog.LogRecord).Attributes())
+			}
+		},
+	}
+}
+
+func accessAttributesKey(mapKey *string) pathGetSetter {
+	return pathGetSetter{
+		getter: func(ctx common.TransformContext) interface{} {
+			return getAttr(ctx.GetItem().(plog.LogRecord).Attributes(), *mapKey)
+		},
+		setter: func(ctx common.TransformContext, val interface{}) {
+			setAttr(ctx.GetItem().(plog.LogRecord).Attributes(), *mapKey, val)
+		},
+	}
+}
+
+func accessDroppedAttributesCount() pathGetSetter {
+	return pathGetSetter{
+		getter: func(ctx common.TransformContext) interface{} {
+			return ctx.GetItem().(plog.LogRecord).DroppedAttributesCount()
+		},
+		setter: func(ctx common.TransformContext, val interface{}) {
+			if i, ok := val.(int64); ok {
+				ctx.GetItem().(plog.LogRecord).SetDroppedAttributesCount(uint32(i))
+			}
+		},
+	}
+}
+
+func accessFlags() pathGetSetter {
+	return pathGetSetter{
+		getter: func(ctx common.TransformContext) interface{} {
+			return ctx.GetItem().(plog.LogRecord).Flags()
+		},
+		setter: func(ctx common.TransformContext, val interface{}) {
+			if i, ok := val.(int64); ok {
+				ctx.GetItem().(plog.LogRecord).SetFlags(uint32(i))
+			}
+		},
+	}
+}
+
+func accessTraceID() pathGetSetter {
+	return pathGetSetter{
+		getter: func(ctx common.TransformContext) interface{} {
+			return ctx.GetItem().(plog.LogRecord).TraceID()
+		},
+		setter: func(ctx common.TransformContext, val interface{}) {
+			if str, ok := val.(string); ok {
+				id, _ := hex.DecodeString(str)
+				var idArr [16]byte
+				copy(idArr[:16], id)
+				ctx.GetItem().(plog.LogRecord).SetTraceID(pcommon.NewTraceID(idArr))
+			}
+		},
+	}
+}
+
+func accessSpanID() pathGetSetter {
+	return pathGetSetter{
+		getter: func(ctx common.TransformContext) interface{} {
+			return ctx.GetItem().(plog.LogRecord).SpanID()
+		},
+		setter: func(ctx common.TransformContext, val interface{}) {
+			if str, ok := val.(string); ok {
+				id, _ := hex.DecodeString(str)
+				var idArr [8]byte
+				copy(idArr[:8], id)
+				ctx.GetItem().(plog.LogRecord).SetSpanID(pcommon.NewSpanID(idArr))
+			}
+		},
+	}
+}
+
+func getAttr(attrs pcommon.Map, mapKey string) interface{} {
```

Comment:
Logic for dealing with resource, IL, attributes are copied directly from traces. I thought it would be easier to review to copy first in this PR and extract to common in another